### PR TITLE
docs: translate API Reference > APIs > `lazy`

### LINKED ISSUE #471: `lazy`
Page: https://id.react.dev/reference/react/lazy

Assignee: @r17x

### PR DESCRIPTION
**Pull request by r17x**
Translation for page https://id.react.dev/reference/react/lazy

Closes #471

### CODE REVIEW

File: src/content/reference/react/lazy.md [hunk 1]
```diff
@@ -16,63 +16,63 @@ const SomeComponent = lazy(load)
...
+* `load`: Sebuah fungsi yang mengembalikan [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) atau *thenable* lain (sebuah objek yang mirip dengan *Promise* dan memiliki metode `then`). React tidak akan memanggil `load` sempai pertama kali And mencoba untuk merender komponen yang dikembalikan. Setelah React pertama kali memanggil `load`, React akan menunggu sampai komponen itu selesai, dan kemudian merender nilai yang telah diselesaikan sebagai komponen React. Baik *Promise* yang dikembalikan maupun nilai yang diselesaikan dari *Promise* akan dicache, sehingga React tidak akan memanggil `load` lebih dari satu kali. Jika *Promise* menolak, React akan `throw` alasan penolakan ke *Error Boundary* terdekat untuk ditangani.
```
Comment (mazipan) [inline src/content/reference/react/lazy.md:35]: Typo sempai, harusnya sampai.
Typo Anda
Nambahin reject, biar tau yang dimaksud menolak itu adalah reject nya Promise.

```suggestion
* `load`: Sebuah fungsi yang mengembalikan [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) atau *thenable* lain (sebuah objek yang mirip dengan *Promise* dan memiliki metode `then`). React tidak akan memanggil `load` sampai pertama kali Anda mencoba untuk merender komponen yang dikembalikan. Setelah React pertama kali memanggil `load`, React akan menunggu sampai komponen itu selesai, dan kemudian merender nilai yang telah diselesaikan sebagai komponen React. Baik *Promise* yang dikembalikan maupun nilai yang diselesaikan dari *Promise* akan dicache, sehingga React tidak akan memanggil `load` lebih dari satu kali. Jika *Promise* menolak (reject), React akan `throw` alasan penolakan ke *Error Boundary* terdekat untuk ditangani.
```

File: src/content/reference/react/lazy.md [hunk 2]
```diff
@@ -16,63 +16,63 @@ const SomeComponent = lazy(load)
...
 
 #### Returns {/*returns*/}
 
-`lazy` returns a React component you can render in your tree. While the code for the lazy component is still loading, attempting to render it will *suspend.* Use [`<Suspense>`](/reference/react/Suspense) to display a loading indicator while it's loading.
+`lazy` mengembalikan kompon React yang dapat Anda render di dalam *tree*. Ketika kode untuk komponen *lazy* masih dimuat, mencoba merendernya akan *suspend.* Gunakan [`<Suspense>`](/reference/react/Suspense) untuk menampilkan indikator pemuatan ketika komponen tersebut dimuat.
```
Comment (mazipan) [inline src/content/reference/react/lazy.md:39]: Typo

```suggestion
`lazy` mengembalikan komponen React yang dapat Anda render di dalam *tree*. Ketika kode untuk komponen *lazy* masih dimuat, mencoba merendernya akan *suspend.* Gunakan [`<Suspense>`](/reference/react/Suspense) untuk menampilkan indikator pemuatan ketika komponen tersebut dimuat.
```

File: src/content/reference/react/lazy.md [hunk 3]
```diff
@@ -16,63 +16,63 @@ const SomeComponent = lazy(load)
...
 
-Usually, you import components with the static [`import`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) declaration:
+Biasanya, Anda mengimpor komponen dengan deklarasi statis [`import`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) :
 
 ```js
 import MarkdownPreview from './MarkdownPreview.js';
 ```
 
-To defer loading this component's code until it's rendered for the first time, replace this import with:
+Untuk menunda pemuatan jkode komponen ini hingga dirender untuk pertama kalinya, ganti *import* ini dengan:
```
Comment (mazipan) [inline src/content/reference/react/lazy.md:65]: Typo
```suggestion
Untuk menunda pemuatan kode komponen ini hingga dirender untuk pertama kalinya, ganti *import* ini dengan:
```

File: src/content/reference/react/lazy.md [hunk 4]
```diff
@@ -16,63 +16,63 @@ const SomeComponent = lazy(load)
...
 import { lazy } from 'react';
 
 const MarkdownPreview = lazy(() => import('./MarkdownPreview.js'));
 ```
 
-This code relies on [dynamic `import()`,](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import) which might require support from your bundler or framework.
+Kode ini bergantung pada [dynamic `import()`,](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import) yang mungkin memerlukan dukungan dari *bundler* atau *framework* yang Anda gunakan.
 
-Now that your component's code loads on demand, you also need to specify what should be displayed while it is loading. You can do this by wrapping the lazy component or any of its parents into a [`<Suspense>`](/reference/react/Suspense) boundary:
+Setelah kode komponen Anda dimuat saat digunakan, Anda juga perlu menentukan apa yang harus ditampilkan ketika dimuat. Anda dapat melakukan ini dengan membungkus komponen *lazy* atau salah satu *parent* ke dalam [`<Suspense>`](/reference/react/Suspense):
```
Comment (mazipan) [inline src/content/reference/react/lazy.md:75]:
```suggestion
Setelah kode komponen Anda dimuat saat digunakan (*on demand*), Anda juga perlu menentukan apa yang harus ditampilkan ketika dimuat. Anda dapat melakukan ini dengan membungkus komponen *lazy* atau salah satu *parent* ke dalam [`<Suspense>`](/reference/react/Suspense):
```

File: src/content/reference/react/lazy.md [hunk 5]
```diff
@@ -175,17 +175,17 @@ body {
...
 ---
 
-## Troubleshooting {/*troubleshooting*/}
+## Pemecahan Masalah {/*troubleshooting*/}
 
-### My `lazy` component's state gets reset unexpectedly {/*my-lazy-components-state-gets-reset-unexpectedly*/}
+### `lazy` komponen saya disetel ulang secara tidak terduka {/*my-lazy-components-state-gets-reset-unexpectedly*/}
 
-Do not declare `lazy` components *inside* other components:
+Jangan deklarasikan `lazy` komponen *didalam* komponen lain:
```
Comment (mazipan) [inline src/content/reference/react/lazy.md:188]:
```suggestion
Jangan deklarasikan `lazy` komponen *di dalam* komponen lain:
```

**Review by mazipan — APPROVED** [review]

File: src/content/reference/react/lazy.md [hunk 6]
```diff
@@ -175,17 +175,17 @@ body {
...
-[Learn more about managing loading states with Suspense.](/reference/react/Suspense)
+[Pelajari lebih lanjut tentang mengelola status pemuatan dengan Suspense.](/reference/react/Suspense)
 
 ---
 
-## Troubleshooting {/*troubleshooting*/}
+## Pemecahan Masalah {/*troubleshooting*/}
 
-### My `lazy` component's state gets reset unexpectedly {/*my-lazy-components-state-gets-reset-unexpectedly*/}
+### `lazy` komponen saya disetel ulang secara tidak terduga {/*my-lazy-components-state-gets-reset-unexpectedly*/}
```
Comment (resir014) [inline src/content/reference/react/lazy.md:186]:
```suggestion
### komponen `lazy` saya disetel ulang secara tidak terduga {/*my-lazy-components-state-gets-reset-unexpectedly*/}
```

File: src/content/reference/react/lazy.md [hunk 7]
```diff
@@ -175,17 +175,17 @@ body {
...
 
-Do not declare `lazy` components *inside* other components:
+Jangan deklarasikan `lazy` komponen *di dalam* komponen lain:
```
Comment (resir014) [inline src/content/reference/react/lazy.md:188]:
```suggestion
Jangan deklarasikan komponen `lazy` *di dalam* komponen lain:
```

File: src/content/reference/react/lazy.md [hunk 8]
```diff
@@ -16,63 +16,63 @@ const SomeComponent = lazy(load)
...
 import { lazy } from 'react';
 
 const MarkdownPreview = lazy(() => import('./MarkdownPreview.js'));
 ```
 
-This code relies on [dynamic `import()`,](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import) which might require support from your bundler or framework.
+Kode ini bergantung pada [dynamic `import()`,](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import) yang mungkin memerlukan dukungan dari *bundler* atau *framework* yang Anda gunakan.
 
-Now that your component's code loads on demand, you also need to specify what should be displayed while it is loading. You can do this by wrapping the lazy component or any of its parents into a [`<Suspense>`](/reference/react/Suspense) boundary:
+Setelah kode komponen Anda dimuat saat digunakan (*on demand*), Anda juga perlu menentukan apa yang harus ditampilkan ketika dimuat. Anda dapat melakukan ini dengan membungkus komponen *lazy* atau salah satu *parent* ke dalam [`<Suspense>`](/reference/react/Suspense):
```
Comment (resir014) [inline src/content/reference/react/lazy.md:75]:
```suggestion
Setelah kode komponen Anda dimuat saat digunakan (*on demand*), Anda juga perlu menentukan apa yang harus ditampilkan ketika dimuat. Anda dapat melakukan ini dengan membungkus komponen *lazy* atau salah satu induknya ke dalam [`<Suspense>`](/reference/react/Suspense):
```

parent component -> komponen induk

**Review by resir014 — CHANGES_REQUESTED** [review]
Some required changes:

**Comment by r17x** [comment]
Need review from you @resir014

**Review by mhaidarhanif — APPROVED** [review]
It's good. @resir014 can confirm.

**Review by mazipan — APPROVED** [review]

**Comment by r17x** [comment]
Waiting for @resir014 approval

**Review by mazipan — APPROVED** [review]

File: src/content/reference/react/lazy.md [hunk 9]
```diff
@@ -4,7 +4,7 @@ title: lazy
 
 <Intro>
 
-`lazy` lets you defer loading component's code until it is rendered for the first time.
+`lazy` memungkinkan Anda menunda pemuatan kode komponen hingga komponen tersebut di-*render* untuk pertama kalinya.
```
Comment (mhaidarhanif) [inline src/content/reference/react/lazy.md:7]: How about this? Continuing from https://github.com/reactjs/id.react.dev/discussions/337#discussioncomment-5759429

```suggestion
`lazy` memungkinkan Anda menangguhkan (*defer*) pemuatan kode komponen hingga komponen tersebut di-*render* untuk pertama kalinya.
```

**Review by mhaidarhanif — CHANGES_REQUESTED** [review]
Suggestion.

File: src/content/reference/react/lazy.md (hunk 9 above)
Comment (r17x) [reply src/content/reference/react/lazy.md]: Applied @mhaidarhanif

**Review by mhaidarhanif — APPROVED** [review]

**Comment by mhaidarhanif** [comment]
Okay let's wait again for @resir014

File: src/content/reference/react/lazy.md [hunk 10]
```diff
@@ -16,63 +16,63 @@ const SomeComponent = lazy(load)
...
 
 #### Returns {/*returns*/}
 
-`lazy` returns a React component you can render in your tree. While the code for the lazy component is still loading, attempting to render it will *suspend.* Use [`<Suspense>`](/reference/react/Suspense) to display a loading indicator while it's loading.
+`lazy` mengembalikan komponen React yang dapat Anda *render* di dalam *tree*. Ketika kode untuk komponen *lazy* masih dimuat, mencoba me-*render*nya akan *suspend.* Gunakan [`<Suspense>`](/reference/react/Suspense) untuk menampilkan indikator pemuatan ketika komponen tersebut dimuat.
 
 ---
 
-### `load` function {/*load*/}
+### fungsi `load` {/*load*/}
```
Comment (resir014) [inline src/content/reference/react/lazy.md:43]:
```suggestion
### Fungsi `load` {/*load*/}
```

File: src/content/reference/react/lazy.md [hunk 11]
```diff
@@ -16,63 +16,63 @@ const SomeComponent = lazy(load)
...
 ```js
 import { lazy } from 'react';
 
 const MarkdownPreview = lazy(() => import('./MarkdownPreview.js'));
 ```
 
-[See more examples below.](#usage)
+[Lihat contoh-contoh lainnya di bawah ini.](#usage)
 
 #### Parameters {/*parameters*/}
```
Comment (resir014) [inline src/content/reference/react/lazy.md:33]:
```suggestion
#### Parameter {/*parameters*/}
```

Suggested change for API reference headings, see: https://github.com/reactjs/id.react.dev/discussions/337#discussioncomment-5769320

File: src/content/reference/react/lazy.md [hunk 12]
```diff
@@ -16,63 +16,63 @@ const SomeComponent = lazy(load)
...
+* `load`: Sebuah fungsi yang mengembalikan [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) atau *thenable* lain (sebuah objek yang mirip dengan *Promise* dan memiliki metode `then`). React tidak akan memanggil `load` sampai pertama kali Anda mencoba untuk me-*render* komponen yang dikembalikan. Setelah React pertama kali memanggil `load`, React akan menunggu sampai komponen itu selesai, dan kemudian me-*render* nilai yang telah diselesaikan sebagai komponen React. Baik *Promise* yang dikembalikan maupun nilai yang diselesaikan dari *Promise* akan dicache, sehingga React tidak akan memanggil `load` lebih dari satu kali. Jika *Promise* menolak (`reject`), React akan melempar (`throw`) alasan penolakan ke *Error Boundary* terdekat untuk ditangani.
 
 #### Returns {/*returns*/}
```
Comment (resir014) [inline src/content/reference/react/lazy.md:37]:
```suggestion
#### Mengembalikan {/*returns*/}
```

Idem.

File: src/content/reference/react/lazy.md [hunk 13]
```diff
@@ -16,63 +16,63 @@ const SomeComponent = lazy(load)
...
 
-`lazy` returns a React component you can render in your tree. While the code for the lazy component is still loading, attempting to render it will *suspend.* Use [`<Suspense>`](/reference/react/Suspense) to display a loading indicator while it's loading.
+`lazy` mengembalikan komponen React yang dapat Anda *render* di dalam *tree*. Ketika kode untuk komponen *lazy* masih dimuat, mencoba me-*render*nya akan *suspend.* Gunakan [`<Suspense>`](/reference/react/Suspense) untuk menampilkan indikator pemuatan ketika komponen tersebut dimuat.
 
 ---
 
-### `load` function {/*load*/}
+### fungsi `load` {/*load*/}
 
 #### Parameters {/*load-parameters*/}
```
Comment (resir014) [inline src/content/reference/react/lazy.md:45]:
```suggestion
#### Parameter {/*load-parameters*/}
```

File: src/content/reference/react/lazy.md [hunk 14]
```diff
@@ -16,63 +16,63 @@ const SomeComponent = lazy(load)
...
 
-`load` receives no parameters.
+`load` tidak menerima parameter.
 
 #### Returns {/*load-returns*/}
```
Comment (resir014) [inline src/content/reference/react/lazy.md:49]:
```suggestion
#### Mengembalikan {/*load-returns*/}
```

File: src/content/reference/react/lazy.md [hunk 15]
```diff
@@ -175,34 +175,34 @@ body {
...
-[Learn more about managing loading states with Suspense.](/reference/react/Suspense)
+[Pelajari lebih lanjut tentang mengelola status pemuatan dengan Suspense.](/reference/react/Suspense)
 
 ---
 
-## Troubleshooting {/*troubleshooting*/}
+## Pemecahan Masalah {/*troubleshooting*/}
 
-### My `lazy` component's state gets reset unexpectedly {/*my-lazy-components-state-gets-reset-unexpectedly*/}
+### komponen `lazy` saya disetel ulang secara tidak terduga {/*my-lazy-components-state-gets-reset-unexpectedly*/}
```
Comment (resir014) [inline src/content/reference/react/lazy.md:186]:
```suggestion
### Komponen `lazy` saya disetel ulang secara tidak terduga {/*my-lazy-components-state-gets-reset-unexpectedly*/}
```

**Review by resir014 — CHANGES_REQUESTED** [review]
@r17x some suggested changes according to https://github.com/reactjs/id.react.dev/discussions/337#discussioncomment-5769320

**Comment by r17x** [comment]
Updated @resir014

File: src/content/reference/react/lazy.md [hunk 16]
```diff
@@ -16,63 +16,63 @@ const SomeComponent = lazy(load)
...
+* `load`: Sebuah fungsi yang mengembalikan [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) atau *thenable* lain (sebuah objek yang mirip dengan *Promise* dan memiliki metode `then`). React tidak akan memanggil `load` sampai pertama kali Anda mencoba untuk me-*render* komponen yang dikembalikan. Setelah React pertama kali memanggil `load`, React akan menunggu sampai komponen itu selesai, dan kemudian me-*render* nilai yang telah diselesaikan sebagai komponen React. Baik *Promise* yang dikembalikan maupun nilai yang diselesaikan dari *Promise* akan dicache, sehingga React tidak akan memanggil `load` lebih dari satu kali. Jika *Promise* menolak (`reject`), React akan melempar (`throw`) alasan penolakan ke *Error Boundary* terdekat untuk ditangani.
 
-#### Returns {/*returns*/}
+#### Mengembalikan {/*returns*/}
```
Comment (mazipan) [inline src/content/reference/react/lazy.md:37]:
```suggestion
#### Kembalian {/*returns*/}
```

Comment (r17x) [reply src/content/reference/react/lazy.md:37]: Wait, What I need to follow 😂 

https://github.com/reactjs/id.react.dev/pull/342#discussion_r1181398261

Comment (mazipan) [reply src/content/reference/react/lazy.md:37]: https://github.com/reactjs/id.react.dev/discussions/337#discussioncomment-5769350

Comment (r17x) [reply src/content/reference/react/lazy.md]: Done @mazipan @resir014

**Review by mazipan — APPROVED** [review]
Looks good

**Comment by r17x** [comment]
Need review @resir014

**Review by resir014 — APPROVED** [review]

**Pull request merged**